pycontrails/datalib/ecmwf/arco_era5.py

```diff
@@ -0,0 +1,579 @@
+
+    See Also
+    --------
+    :func:`open_arco_era5_model_level_data`
+    :func:`open_arco_era5_single_level`
+    """
+
+    grid: float
+
+    __marker = object()
+
+    def __init__(
+        self,
+        time: datalib.TimeInput,
+        variables: datalib.VariableInput,
+        pressure_levels: datalib.PressureLevelInput | None = None,
+        grid: float = 0.25,
+        cachestore: cache.CacheStore | None = __marker,  # type: ignore[assignment]
+        n_jobs: int = 1,
+        cleanup_metview_tempfiles: bool = True,
+    ) -> None:
+        self.timesteps = datalib.parse_timesteps(time)
+
+        if pressure_levels is None:
+            self.pressure_levels = pressure_levels_at_model_levels(20_000.0, 50_000.0)
+        else:
+            self.pressure_levels = datalib.parse_pressure_levels(pressure_levels)
+
+        self.paths = None
+        self.variables = datalib.parse_variables(variables, self.supported_variables)
+        self.grid = grid
+        self.cachestore = cache.DiskCacheStore() if cachestore is self.__marker else cachestore
+        self.n_jobs = n_jobs
+        self.cleanup_metview_tempfiles = cleanup_metview_tempfiles
+
+    @property
+    def pressure_level_variables(self) -> list[met_var.MetVariable]:
+        """Variables available in the ARCO ERA5 model level data.
+
+        Returns
+        -------
+        list[MetVariable] | None
+            List of MetVariable available in datasource
+        """
+        return PRESSURE_LEVEL_VARIABLES
+
+    @property
+    def single_level_variables(self) -> list[met_var.MetVariable]:
+        """Variables available in the ARCO ERA5 single level data.
+
+        Returns
+        -------
+        list[MetVariable] | None
+            List of MetVariable available in datasource
+        """
+        return ecmwf_variables.SURFACE_VARIABLES
+
+    @overrides
+    def download_dataset(self, times: list[datetime.datetime]) -> None:
+        if not times:
+            return
+
+        # Download single level data sequentially
+        if self.is_single_level:
+            unique_dates = sorted({t.date() for t in times})
+            for t in unique_dates:
+                ds = open_arco_era5_single_level(t, self.variables)
+                self.cache_dataset(ds)
+            return
+
+        stack = contextlib.ExitStack()
+        if self.cleanup_metview_tempfiles:
+            stack.enter_context(_MetviewTempfileHandler())
+
+        n_jobs = min(self.n_jobs, len(times))
+
+        # Download sequentially if n_jobs == 1
+        if n_jobs == 1:
+            for t in times:
+                with stack:  # cleanup after each iteration
+                    _download_convert_cache_handler(self, t)
+            return
+
+        # Download in parallel
+        args = [(self, t) for t in times]
+        mp = multiprocessing.get_context("spawn")
+        with mp.Pool(n_jobs) as pool, stack:  # cleanup after pool is closed
```

Comment:
I think you want 
```python
with stack, mp.Pool(n_jobs) as pool:
```
if it's important to clean up temporary files only after the pool is closed (https://stackoverflow.com/questions/68248141/is-the-order-of-multiple-exit-calls-in-a-with-block-always-deterministic).

Cleanup with `n_jobs > 1` is working well for me, though, so this may not be an issue.